python/node_similarity.py

```diff
@@ -0,0 +1,289 @@
+from math import sqrt
+from typing import Set, List, Callable
+from itertools import product
+
+import mgp
+
+
+# not implemented for directed graphs
+# not implemented for multigraphs
+# not taking into account edge weights
+
+@mgp.read_proc
+def jaccard(
+        context: mgp.ProcCtx,
+        node1: mgp.Any,
+        node2: mgp.Any,
+        mode: str = 'c'
```

Comment:
Although this is way shorter, maybe we should allow both 'c' and 'cartesian' as arguments?